### PR TITLE
Docs: Add basic query option descriptions 

### LINKED ISSUE #66291: Need basic query option descriptions in documentation
**What happened**:
Frustrated user emailed docs with the following:
"It's been hours trying to create a query variable with "display text" and "value", the value to be used in the dashboard for filtering!

We have read [this](https://grafana.com/docs/grafana/latest/dashboards/variables/add-template-variables/#add-a-query-variable) document tons of times, and we tried "SELECT name AS text, id AS value" with no luck, after hours, we find a small section [here](https://community.grafana.com/t/variable-query-options-text-value-in-sql/76887) mentioned the right field names! how do you guys expect people to find the reserved columns names "__text" and "__value" without mentioning this in the documentation?!

At least! mention that as a hint in the "Query options" section in Grafana!!"

**What's not clear**
I don't know if this is a PromQL issue, and we need to do a better job of linking out to their documentation, or if our own docs need to be expanded. Could also be a basic SQL thing that I'm totally unaware of.

### PR DESCRIPTION
**Pull request by moxious**
Fixes https://github.com/grafana/grafana/issues/66291

User on that ticket had a really good point.  Admonitions added, example given.  Should make this page easier / less frustrating to use.

### CODE REVIEW

File: docs/sources/dashboards/variables/add-template-variables/index.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ Query variables are generally only supported for strings. If your query returns
 Query expressions can contain references to other variables and in effect create linked variables. Grafana detects this and automatically refreshes a variable when one of its linked variables change.
 
 {{% admonition type="note" %}}
-Query expressions are different for each data source. For more information, refer to the documentation for your [data source]({{< relref "../../../datasources/" >}}).
+The query should return two fields named `__text` and `__value`. The `__text` column value should be unique (if it is not unique then the first value is used). Query expressions are different for each data source. For more information, refer to the documentation for your [data source]({{< relref "../../../datasources/" >}}).
```
Comment (imatwawana) [inline docs/sources/dashboards/variables/add-template-variables/index.md:90]:
```suggestion
The query should return two fields named `__text` and `__value`. The `__text` column value should be unique (if it's not unique, then the first value is used). Query expressions are different for each data source. For more information, refer to the documentation for your [data source]({{< relref "../../../datasources/" >}}).
```

File: docs/sources/dashboards/variables/add-template-variables/index.md [hunk 2]
```diff
@@ -97,6 +97,7 @@ Query expressions are different for each data source. For more information, refe
    - **On Time Range Change:** Queries the data source when the dashboard time range changes. Only use this option if your variable options query contains a time range filter or is dependent on the dashboard time range.
 1. In the **Query** field, enter a query.
    - The query field varies according to your data source. Some data sources have custom query editors.
+   - Make sure that the query returns values named `__text` and `__value` as appropriate in your query syntax. As a SQL-specific example, you can use a query such as `SELECT hostname AS __text, id AS __value FROM MyTable`; other query languages will vary depending on syntax.
```
Comment (imatwawana) [inline docs/sources/dashboards/variables/add-template-variables/index.md:100]: I've taken out the semi-colon and just made a new sentence at the end. But that last sentence doesn't entirely make sense to me. Does my revision work?
```suggestion
   - Make sure that the query returns values named `__text` and `__value` as appropriate in your query syntax. For example, in SQL, you can use a query such as `SELECT hostname AS __text, id AS __value FROM MyTable`. Queries for other languages will vary depending on syntax.
```

**Review by imatwawana — COMMENTED** [review]
Thanks for the additions! I've reviewed for wording and left one question.

File: docs/sources/dashboards/variables/add-template-variables/index.md (hunk 1 above)
Comment (imatwawana) [reply docs/sources/dashboards/variables/add-template-variables/index.md:90]: Looking at the way this admonition is being used currently on the page, it doesn't make sense to me to add this information in the same admonition. Seems like it should be its own content outside of the admonition entirely.

Comment (imatwawana) [inline docs/sources/dashboards/variables/add-template-variables/index.md:90]: Looking at the way this admonition is being used currently on the page, it doesn't make sense to me to add this information in the same admonition. Seems like it should be its own content outside of the admonition entirely. 
```suggestion
Query expressions are different for each data source. For more information, refer to the documentation for your [data source]({{< relref "../../../datasources/" >}}).
```

File: docs/sources/dashboards/variables/add-template-variables/index.md [hunk 3]
```diff
@@ -87,7 +87,7 @@ Query variables are generally only supported for strings. If your query returns
 Query expressions can contain references to other variables and in effect create linked variables. Grafana detects this and automatically refreshes a variable when one of its linked variables change.
```
Comment (imatwawana) [inline docs/sources/dashboards/variables/add-template-variables/index.md:88]:
```suggestion

The query should return two fields named `__text` and `__value`. The `__text` column value should be unique. If it 's not unique, then the first value is used. 

```

**Comment by moxious** [comment]
Revisions are good to go.  I think the sentence does make sense as you've re-written it.  An unfortunate complexity of this is that query variables can use any query language.  SQL was given as probably the most common example people will use, but it's not an all-encompassing example, since any of 5 other query languages could be needed, and how you'd express the field would be different in each.   That built-in complexity isn't fixable here though, it's just how grafana works

**Review by imatwawana — APPROVED** [review]

**Pull request merged**